### ISSUE TITLE
Segfault when running FFT performance test suite

### ISSUE THREAD

**Issue by stewdk** [projectNe10/Ne10#154]
I'm getting a segfault due to accessing the ne10_log_buffer (via ne10_log_buffer_ptr) out of bounds.  I attached a bandaid (patch) that I used to get around it, but perhaps the usage of this buffer should be further evaluated.
[fft_perf_test_segfault_fix.patch.txt](https://github.com/projectNe10/Ne10/files/774294/fft_perf_test_segfault_fix.patch.txt)

**Comment by joesavage** [comment]
Hmm... that's weird. The initialisation of `ne10_log_buffer_ptr` should always occur within the unit test setup procedure — certainly that appears to be the case in the INT32 FFT test, which seems to be working correctly on my machine. Would you be able to provide any more details about which part of the project you're building/running, and how you're going about this?

**Comment by stewdk** [comment]
You're right, it gets initialized at the start of the test setup, but the pointer eventually gets incremented past the upper bound of ne10_log_buffer.  My patch just resets the pointer back to the start of the buffer more often. An alternative fix might be to increase the size of the buffer, but that wouldn't necessarily be favorable for benchmarking memory-limited systems. The segfault occurs at line 398 of test_suite_fft_int32.c while running an FFT of size 32768.
Use this patch to see exactly when the pointer goes out of bounds.
[ne10_print_outofbounds.patch.txt](https://github.com/projectNe10/Ne10/files/774969/ne10_print_outofbounds.patch.txt)

**Comment by stewdk** [comment]
To fully answer your question, I'm running it on aarch64 and I used the following cmake command line to build the project: `cmake -DCMAKE_TOOLCHAIN_FILE=../GNUlinux_config.cmake -DNE10_BUILD_UNIT_TEST=ON -DNE10_PERFORMANCE_TEST=ON ..`

**Comment by joesavage** [comment]
Ah, gotcha — sorry, I misunderstood. It looks like this is related to #113. Effectively, it seems like even though `ne10_log` doesn't need to keep its output around in one big buffer for its common use cases (e.g. when running the tests normally, `ne10_log_buffer_ptr` need not change value), this functionality *is* used within the iOS and Android demo applications (where the test log output is manipulated for some reason or another).

For the time being, using a bigger buffer here should resolve the issue, but it seems like in the longer term the demo applications should be changed such that they don't rely on the unit tests lugging around their log output.
